aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -211,12 +211,625 @@ Tensor& replication_pad1d_backward_out_cpu_template(
+            pad_l, pad_r,
+            pad_t, pad_b);
+        }
+        );
+  }
+  else
+  {
+    int64_t p;
+
+    output.resize_({nbatch, nslices, oheight, owidth});
+
+#pragma omp parallel for private(p)
+    for (p = 0; p < nbatch; p++)
+    {
+      AT_DISPATCH_FLOATING_TYPES(input_.type(), "replication_pad2d", [&] {
+          auto input_data = input_.data<scalar_t>();
+          auto output_data = output.data<scalar_t>();
+          replication_pad2d_out_frame<scalar_t>(
+              input_data+p*nslices*iwidth*iheight,
+              output_data+p*nslices*owidth*oheight,
+              nslices,
+              iwidth, iheight,
+              owidth, oheight,
+              pad_l, pad_r,
+              pad_t, pad_b);
+          }
+          );
+    }
+  }
+}
+
+template <typename scalar_t>
+static void replication_pad2d_backward_out_frame(
+    scalar_t *ginput_p, scalar_t *goutput_p,
+    int64_t nslices,
+    int64_t iwidth, int64_t iheight,
+    int64_t owidth, int64_t oheight,
+    int pad_l, int pad_r,
+    int pad_t, int pad_b)
+{
+  int iStartX = fmax(0, -pad_l);
+  int iStartY = fmax(0, -pad_t);
+  int oStartX = fmax(0, pad_l);
+  int oStartY = fmax(0, pad_t);
+
+  int64_t k, ip_x, ip_y;
+#pragma omp parallel for private(k, ip_x, ip_y)
+  for (k = 0; k < nslices; k++)
+  {
+    int64_t i, j;
+    for (i = 0; i < oheight; i++) {
+      for (j = 0; j < owidth; j++) {
+        if (j < pad_l) {
+          ip_x = pad_l;
+        } else if (j >= pad_l && j < iwidth + pad_l) {
+          ip_x = j;
+        } else {
+          ip_x = iwidth + pad_l - 1;
+        }
+        ip_x = ip_x - oStartX + iStartX;
+
+        if (i < pad_t) {
+          ip_y = pad_t;
+        } else if (i >= pad_t && i < iheight + pad_t) {
+          ip_y = i;
+        } else {
+          ip_y = iheight + pad_t - 1;
+        }
+        ip_y = ip_y - oStartY + iStartY;
+
+        scalar_t *src_p = goutput_p + k*owidth*oheight + i * owidth + j;
+        scalar_t *dest_p = ginput_p + k*iwidth*iheight + ip_y * iwidth + ip_x;
+        *dest_p += *src_p;
+      }
+    }
+  }
+}
+
+Tensor& replication_pad2d_backward_out_cpu_template(
+    Tensor& gradInput,
+    const Tensor& gradOutput_,
+    const Tensor& input,
+    IntList paddingSize) 
+{
+  int pad_l = paddingSize[0];
+  int pad_r = paddingSize[1];
+  int pad_t = paddingSize[2];
+  int pad_b = paddingSize[3];
+  int dimw = 2;
+  int dimh = 1;
+  int dimslices = 0;
+  int64_t nbatch = 1;
+  int64_t nslices;
+  int64_t iheight;
+  int64_t iwidth;
+  int64_t oheight;
+  int64_t owidth;
+
+  if (input.dim() == 4)
+  {
+    nbatch = input.size(0);
+    dimw++;
+    dimh++;
+    dimslices++;
+  }
+
+  /* sizes */
+  nslices = input.size(dimslices);
+  iheight = input.size(dimh);
+  iwidth = input.size(dimw);
+  oheight = iheight + pad_t + pad_b;
+  owidth  = iwidth + pad_l + pad_r;
+
+  AT_CHECK(owidth == gradOutput_.size(dimw),
+      "gradOutput width unexpected. Expected: ", owidth, ", Got: ",
```

Comment:
I tested, there should be no space automatically added after the comma. see the output of "inputtest", it would be "input test" if there is a space.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lhuang04/local/miniconda/lib/python3.7/site-packages/torch/nn/functional.py", line 2686, in pad
    ret = torch._C._nn.replication_pad1d(input, pad)
RuntimeError: non-empty 2D or 3D (batch mode) tensor expected for inputtest

 56   AT_CHECK(input.numel() > 0
 57       && (input.ndimension() == 2 || input.ndimension() == 3),
 58       "non-empty 2D or 3D (batch mode) tensor expected for input", "test");